dune-project

```diff
@@ -1,2 +1,5 @@
-(lang dune 1.0)
+(lang dune 2.0)
```

Comment:
This needs the ppx_cstruct.opam to have a corresponding `dune {>= "2.0.0"}` entry.

Reply:
Done.

Reply:
Sorry, just one last change needed here before merge. Since the `dune-project` affects all the associated projects, every opam package needs a corresponding `dune >2` entry. e.g. the Drone CI is failing as its trying to install `cstruct` with dune 1.11 and that no longer works.

Reply:
I have now added dune {>= "2.0.0"} constraint to all .opam files in repo.